### PR TITLE
Two commits that fix quite a few of libhtml memory leaks

### PR DESCRIPTION
**Pull request by nsajko**

### CODE REVIEW

File: src/libhtml/build.c [hunk 1]
```diff
@@ -2127,13 +2127,17 @@ getitems(ItemSource* is, uchar* data, int datalen)
 	outerps = lastps(ps);
 	ans = outerps->items->next;
 	/* note: ans may be nil and di->kids not nil, if there's a frameset! */
+	if (outerps->items != nil) {
```
Comment (dancrossnyc) [inline src/libhtml/build.c]: Just free without the if please.

File: src/libhtml/lex.c [hunk 2]
```diff
@@ -584,14 +587,22 @@ getplaindata(TokenSource* ts, Token* a, int* pai)
 		if(c != 0){
 			buf[j++] = c;
 			if(j == sizeof(buf)-1){
+				tmpS = s;
 				s = buftostr(s, buf, j);
+				if (tmpS != nil) {
```
Comment (dancrossnyc) [inline src/libhtml/lex.c]: Just `free(tmpS);` here.

File: src/libhtml/lex.c [hunk 3]
```diff
@@ -584,14 +587,22 @@ getplaindata(TokenSource* ts, Token* a, int* pai)
...
+				}
 				j = 0;
 			}
 		}
 		if(c == '\n')
 			break;
 	}
+	tmpS = s;
 	s = buftostr(s, buf, j);
+	if (tmpS != nil) {
```
Comment (dancrossnyc) [inline src/libhtml/lex.c]: Same here. Just the free, no if.

**Comment by dancrossnyc** [comment]
I made a few line comments, but the general point stands: instead of `if(p) { free(p); }` just `free(p);` throughout.

**Comment by nsajko** [comment]
@dancrossnyc done.

File: src/libhtml/lex.c [hunk 4]
```diff
@@ -734,15 +745,19 @@ getscriptdata(TokenSource* ts, int firstc, int starti, Token* a, int* pai)
 		if(c != 0){
 			buf[j++] = c;
 			if(j == BIGBUFSIZE-1){
+				tmpS = s;
```
Comment (dancrossnyc) [inline src/libhtml/lex.c]: This pattern is repeated enough that something like,

```
static Rune*
Buftostr(Rune* s, Rune *buf, int j)
{
        Rune *tmp = s;
        s = buftostr(tmp, buf, j);
        free(tmp);
        return s;
}
```

Might be warranted.

Comment (nsajko) [reply src/libhtml/lex.c]: Yeah, I changed the buftostr API, and now the change is much nicer. (Every call was a memleak.)

**Comment by nsajko** [comment]
@dancrossnyc I think this is OK now?

**Comment by dancrossnyc** [comment]
Indeed it is.

**Pull request merged**